Automattic/jetpack
# renovate: Fix warnings from weird wpcomsh deps

## Proposed changes:

For the five `type: vcs` deps, Renovate needs a `name` key in each one so it knows the package name (versus Composer itself that fetches the composer.json).

The "wordpress/classic-editor-plugin" fake package is a bit more difficult. We have to disable the 'packagist' entry entirely. But then we can make a customManager and customDatasource for it. 😀

### Other information:


## Jetpack product discussion


None

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






If you look at our current [Renovate dashboard](https://github.com/Automattic/jetpack/issues/23363) and recent renovate PRs like https://github.com/Automattic/jetpack/pull/37860, you'll see warnings like

> [!WARNING]
> Renovate failed to look up the following dependencies: `Failed to look up packagist package automattic/at-pressable-podcasting`, `Failed to look up packagist package automattic/custom-fonts`, `Failed to look up packagist package automattic/custom-fonts-typekit`, `Failed to look up packagist package automattic/text-media-widget-styles`, `Failed to look up packagist package automattic/wc-calypso-bridge`, `Failed to look up packagist package wordpress/classic-editor-plugin`.
> 
> Files affected: `projects/plugins/wpcomsh/composer.json`

  and

> [!WARNING]
> Some dependencies could not be looked up. Check the Dependency Dashboard for more information.


Cherry-pick this into a fork, set up the secrets so you can run the Renovate action there, and do so. Those should go away.
  * Or you could look at e.g. https://github.com/anomiex/jetpack/issues/1 and https://github.com/anomiex/jetpack/pull/39 where I already did that. The latter also illustrates that the custom datasource is good enough to support creating upgrade PRs.
